packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```diff
@@ -35,20 +43,44 @@ const initialState: SliceBuilderState = {
   status: null,
 };
 
-const SliceBuilder: React.FunctionComponent = () => {
-  const { Model, store, variation } = useContext(SliceContext);
+interface SliceBuilderProps {
+  initModel: SliceState;
+  initVariationId: string;
+}
 
-  const { openLoginModal, checkSimulatorSetup, openToaster } =
-    useSliceMachineActions();
+const SliceBuilder: React.FC<SliceBuilderProps> = ({
+  initModel,
+  initVariationId,
+}) => {
+  useEffect(() => {
+    initSliceStore(initModel);
+  }, []);
+
+  const { Model } = useSelector((state: SliceMachineStoreType) => ({
```

Comment:
I just saw the `SliceBuilderWithRouter` I think we don't need to retrieve the model here as it's the same in the props.

And if the store changes, I believe the whole `SliceBuilderWithRouter` will re-render and so re-render this one.

Reply:
oh right!!! nice one for spotting